src/applications/lifecycles/unmount.js

```diff
@@ -1,20 +1,39 @@
-import { UNMOUNTING, NOT_MOUNTED, MOUNTED, SKIP_BECAUSE_BROKEN, getAppProps } from '../app.helpers.js';
+import { UNMOUNTING, NOT_MOUNTED, MOUNTED, SKIP_BECAUSE_BROKEN } from '../app.helpers.js';
 import { handleAppError } from '../app-errors.js';
 import { reasonableTime } from '../timeouts.js';
+import { getProps } from './prop.helpers.js';
 
-export async function toUnmountPromise(app) {
-  if (app.status !== MOUNTED) {
-    return app;
+export async function toUnmountPromise(appOrParcel) {
+  if (appOrParcel.status !== MOUNTED) {
+    return appOrParcel;
   }
-  app.status = UNMOUNTING;
+  appOrParcel.status = UNMOUNTING;
+
+  const unmountChildrenParcels = Object.keys(appOrParcel.parcels)
+    .map(parcelId => appOrParcel.parcels[parcelId].unmountThisParcel());
+
+  let parcelError;
 
   try {
-    await reasonableTime(app.unmount(getAppProps(app)), `Unmounting application ${app.name}'`, app.timeouts.unmount);
-    app.status = NOT_MOUNTED;
+    await Promise.all(unmountChildrenParcels);
   } catch (err) {
-    handleAppError(err, app);
-    app.status = SKIP_BECAUSE_BROKEN;
+    parcelError = err;
+    handleAppError(err, appOrParcel);
+    appOrParcel.status = SKIP_BECAUSE_BROKEN;
```

Comment:
Should the parent parcel be `BROKEN` when the child fails to unmount?

Reply:
I don't think so -- SKIP_BECAUSE_BROKEN means that the activity function is never called and that it will never be unmounted/remounted. I guess for parcels, it's a bit different because they do not have activity functions and they are imperatively unmounted/remounted. So maybe SKIP_BECAUSE_BROKEN doesn't really mean much for an individual parcel.

Do you see benefits to putting the owning application into SKIP_BECAUSE_BROKEN when a child parcel throws an error? I feel like that might be harsh on the application -- it's not really its fault that the child parcel died.